packages/visx-axis/src/types.ts

```diff
@@ -19,7 +19,11 @@ export type TickFormatter<T> = (
   values: { value: T; index: number }[],
 ) => FormattedValue;
 
-export type TickLabelProps<T> = (value: T, index: number) => Partial<TextProps>;
+export type TickLabelProps<T> = (
+  value: T,
+  index: number,
+  values: { value: T; index: number }[],
```

Comment:
Does this mean tsc will require folks to update the arguments for every tickLabelProps? guess I'm trying to figure out if a tsc warning/error on this would break builds without everyone going in and updating all of their functions to the new signature

Reply:
Passing function that takes fewer arguments should be ok.
Passing function that requires more argument is not.

```ts
type MyFnType = ((a: string) => string);

// OK
const f1: MyFnType = () => 'a';
// OK
const f2: MyFnType = (a: string) => a;
// error
const f3: MyFnType = (a: string, b: string) => 'a';
```

Reply:
^yeah I don't think that using the old signature will be problematic `(value, index) => TextProps` shouldn't throw. One thing that would break is if folks are using `TickLabelProps` explicitly in their typing:

```ts
// error
(value, index): TickLabelProps<ValueType> => TextProps
```

That would throw because the signature doesn't match (`XYChart` wraps `Axis` and uses that type, so I had to update the signature [on this line](https://github.com/airbnb/visx/pull/1044/files#diff-a1e01e13b948f291db2d3d3506abe738e077f948a5f4fecf2fb543d2e11c5624R37))

I'd guess that would be rare. I don't think we've considered type changes like that breaking historically (e.g., bumping d3 types wasn't breaking even though there were maybe breaking changes)